modules/images/webp-uploads/load.php

```diff
@@ -683,3 +683,120 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
 	return rest_ensure_response( $data );
 }
 add_filter( 'rest_prepare_attachment', 'webp_uploads_update_rest_attachment', 10, 3 );
+
+/**
+ * Adds sources to metadata for an attachment.
+ *
+ * @param array  $metadata              Metadata of the attachment.
+ * @param array  $valid_mime_transforms List of valid mime transforms for current image mime type.
+ * @param array  $allowed_mimes         Allowed mime types.
+ * @param string $file                  Path to original file.
+ * @return array                        Metadata with sources added.
+ */
+function webp_uploads_update_sources( $metadata, $valid_mime_transforms, $allowed_mimes, $file ) {
+	$original_directory = pathinfo( $file, PATHINFO_DIRNAME );
+
+	foreach ( $valid_mime_transforms as $targeted_mime ) {
+		// Add sources to original image metadata.
+		$extension            = explode( '|', $allowed_mimes[ $targeted_mime ] );
+		$filename_without_ext = implode( explode( '.', $file, -1 ) );
```

Comment:
Would be ideal to rely on `pathinfo` instead.

```suggestion
		$filename_without_ext = pathinfo( $file, PATHINFO_FILENAME );
```